sql-cloudant/src/main/scala/org/apache/bahir/cloudant/CloudantAllDocsConfig.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bahir.cloudant
+
+import org.apache.bahir.cloudant.common.JsonStoreConfigManager
+
+class CloudantAllDocsConfig(protocol: String, host: String, dbName: String,
+                            indexName: String = null, viewName: String = null)
+                           (username: String, password: String, partitions: Int,
+                            maxInPartition: Int, minInPartition: Int, requestTimeout: Long,
+                            bulkSize: Int, schemaSampleSize: Int,
+                            createDBOnSave: Boolean, apiReceiver: String, selector: String,
+                            useQuery: Boolean, queryLimit: Int)
+  extends CloudantConfig(protocol, host, dbName, indexName, viewName)(username, password,
+  partitions, maxInPartition, minInPartition, requestTimeout, bulkSize, schemaSampleSize,
+  createDBOnSave, apiReceiver, selector, useQuery, queryLimit) {
+
+  override val defaultIndex: String = apiReceiver
+
```

Comment:
Adding small comments for functions `getOneUrl`, `getUrl`, and `getSubSetUrl` describing how these functions are used would be a great help for understanding the code.